src/with-chatkit-one-to-one.js

```diff
@@ -89,85 +89,97 @@ export function withChatkitOneToOne(WrappedComponent) {
+                  this.setState({
+                    otherUserLastReadMessageId: cursor.position,
+                  })
+                }
               },
-            })
+            },
+          }),
         )
-          .then(() =>
-            this.context.chatkit.currentUser.subscribeToRoomMultipart({
-              roomId: this._roomId,
-              messageLimit: this._messageLimitOnLoad,
-              hooks: {
-                onMessage: message =>
-                  this.setState(state => ({
-                    messages: [...state.messages, message],
-                  })),
-
-                onPresenceChanged: (state, user) => {
-                  if (user.id === this.props.otherUserId) {
-                    this.forceUpdate()
-                  }
-                },
-
-                onUserStartedTyping: user => {
-                  if (user.id === this._otherUserId) {
-                    this.setState({ otherUserIsTyping: true })
-                  }
-                },
-
-                onUserStoppedTyping: user => {
-                  if (user.id === this._otherUserId) {
-                    this.setState({ otherUserIsTyping: false })
-                  }
-                },
-
-                onNewReadCursor: cursor => {
-                  const cursorBelongsToOtherUser =
-                    cursor.user.id === this._otherUserId
-                  if (cursorBelongsToOtherUser) {
-                    this.setState({
-                      otherUserLastReadMessageId: cursor.position,
-                    })
-                  }
-                },
+        .then(room =>
+          this.setState({
+            otherUser: room.users.find(u => u.id === this._otherUserId),
+            otherUserLastReadMessageId: this.context.chatkit.currentUser.readCursor(
+              {
+                userId: this._otherUserId,
+                roomId: this._roomId,
               },
-            }),
-          )
-          .then(room =>
-            this.setState({
-              otherUser: room.users.find(u => u.id === this._otherUserId),
-              otherUserLastReadMessageId: this.context.chatkit.currentUser.readCursor(
-                {
-                  userId: this._otherUserId,
-                  roomId: this._roomId,
-                },
-              ),
-              isLoading: false,
-            }),
-          )
-          .catch(err => console.error(err))
-      })
+            ),
+            isLoading: false,
+          }),
+        )
+        .catch(err => console.error(err))
+    }
+
+    _disconnectFromRoom() {
+      this.context.chatkit.currentUser.roomSubscriptions[this._roomId].cancel()
+    }
+
+    componentDidMount() {
+      this.context.addOnLoadListener(() => this._connectToRoom())
     }
 
     render() {
+      if (!this.context.chatkit.isLoading) {
+        // If the otherUserId prop has changed, connect to a new one-to-one room
+        if (this.props.otherUserId != this._otherUserId) {
```

Comment:
prefer triple equals

Reply:
I'm surprise the linter didn't pick it up :thinking: